src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1331,9 +1336,21 @@ bool isRefEscape
             // widest possible escape via writeable ref-like receiver or ref/out argument.
             uint escapeTo = scopeOfTheContainingExpression;
 
+            var isReceiverRefReadOnly = symbol switch
+            {
```

Comment:
This is computed even when it is not needed/used (it is the second operand of a &&).  Please break it out into a local function and call it only when needed.

Reply:
Done